astro/src/content/docs/operate/deploy/terraform.mdx

```diff
@@ -0,0 +1,432 @@
+
+## Using Data Sources
+
+Instead of importing a resource, you can use a `Data Source`. The list of supported FusionAuth data sources is in the [Terraform Provider documentation](https://registry.terraform.io/providers/fusionauth/fusionauth/latest/docs).
+
+Data sources are useful if you choose to manage the default tenant and FusionAuth application outside of Terraform, either manually or via a script using the [client libraries](/docs/v1/tech/client-libraries/), but you still need to reference the tenant or application. Examples of this include:
+
+* adding applications in the default tenant
+* associating a JWT signing key with the FusionAuth application
+* setting up an IP ACL to limit access to the FusionAuth application
+
+Review the [tenant](https://registry.terraform.io/providers/fusionauth/fusionauth/latest/docs/data-sources/tenant) and [application](https://registry.terraform.io/providers/fusionauth/fusionauth/latest/docs/data-sources/application) data source documentation to learn more.
+
+Here's an example of how you might add data sources to your Terraform file:
+
+<RemoteCode lang="hcl" title="A Data Source For The default tenant" url="https://raw.githubusercontent.com/fusionauth/fusionauth-example-terraform/main/examples/data-source/main.tf" tags="defaultTenantDataSource" />
+
+<RemoteCode lang="hcl" title="A Data Source For The FusionAuth application" url="https://raw.githubusercontent.com/fusionauth/fusionauth-example-terraform/main/examples/data-source/main.tf" tags="defaultApplicationDataSource" />
+
+<Aside type="tip">
+If you manage the default tenant and FusionAuth application outside of Terraform but want specific actions (scripts, API calls, etc.) integrated and triggered by your Terraform configuration, you could use [Provisioners](https://developer.hashicorp.com/terraform/language/resources/provisioners/syntax). Terraform includes the concept of provisioners as a measure of pragmatism and last resort since provisioners are non-declarative and potentially unpredictable.
+</Aside>
+
+Now that you've configured Terraform to handle some of FusionAuth's default configuration elements, let's walk through creating, updating, and removing other resources.
+
+## Creating Resources
+
+To create a new resource review the list of resources available to you in the [FusionAuth Terraform Provider documentation](https://registry.terraform.io/providers/fusionauth/fusionauth/latest/docs). Pick the resource you're interested in. Each resource contains information about required and optional arguments.
+
+For this example, you'll create the following:
+
+* A new tenant called `Forum`.
+* An application living in that new tenant, also called `Forum`.
+* Two roles for that application, `admin` and `user`.
+* A token signing key for that application.
+* Email templates for password emails that will be assigned to the tenant.
+
+### Variables
```

Comment:
It seems odd to define `fusionauth_default_theme_id` as a variable here. Wouldn't one reference the data source or resource of the default tenant directly?

Reply:
I was not considering referencing the theme_id from the tenant configuration as an option, this would be something to be tested. 
And as it would be otherwise a resource you have to create https://registry.terraform.io/providers/gpsinsight/fusionauth/latest/docs/resources/theme and does not exist as datasource I went with a variable instead for this example.

Reply:
I'm going to leave this as a variable because someone could jump directly to the create section and not import or add the default tenant as a data source. I'll make a note, though.